projects/pgai/pgai/cli.py

```diff
@@ -219,13 +223,23 @@ def vectorizer_worker(
         if pgai_version is None:
             log.critical("the pgai extension is not installed")
             sys.exit(1)
-        vectorizer_ids = get_vectorizer_ids(cur, vectorizer_ids)
-    if len(vectorizer_ids) == 0:
-        log.warning("no vectorizers found")
-        return
+    dynamic_mode = len(vectorizer_ids) == 0
+    valid_vectorizer_ids = []
+    if not dynamic_mode:
+        valid_vectorizer_ids = get_vectorizer_ids(db_url, vectorizer_ids)
+        if len(valid_vectorizer_ids) != len(vectorizer_ids):
+            log.critical(
+                f"invalid vectorizers, wanted: {list(vectorizer_ids)}, got: {valid_vectorizer_ids}"  # noqa: E501 (line too long)
+            )
+            sys.exit(1)
 
     while True:
-        for vectorizer_id in vectorizer_ids:
+        if dynamic_mode:
+            valid_vectorizer_ids = get_vectorizer_ids(db_url, vectorizer_ids)
+            if len(valid_vectorizer_ids) == 0:
+                log.warning("no vectorizers found")
```

Comment:
This is nice, because it also stops it from crashing when there is no vectorizer defined yet.